crates/holo_hash/src/hash.rs

```diff
@@ -111,15 +174,14 @@ fn bytes_to_loc(bytes: &[u8]) -> u32 {
         + ((bytes[3] as u32) << 24)
 }
 
-fn assert_length(hash: &[u8]) {
-    if hash.len() != HOLO_HASH_SERIALIZED_LEN {
-        panic!(
-            "invalid holo_hash byte count, expected: {}, found: {}. {:?}",
-            HOLO_HASH_SERIALIZED_LEN,
-            hash.len(),
-            hash
-        );
-    }
+/// Helper for ensuring the the proper number of bytes is used in various situations
+pub fn assert_length(len: usize, hash: &[u8]) {
+    debug_assert_eq!(
+        hash.len(),
+        len,
+        "invalid byte count for HoloHash {:?}",
+        hash
+    );
 }
```

Comment:
I assume the compiler will optimize out an empty function... but, just in case, we can make sure this doesn't even make it to that stage via:

```suggestion
macro_rules! assert_length {
    ($len:literal, $hash:expr) => {
        debug_assert_eq!(
            $hash.len(),
            $len,
            "invalid byte count for HoloHash {:?}",
            $hash
    );
    };
}
```